### DISCUSSION TITLE
Extract raw SQL from object

### DISCUSSION THREAD

**Discussion by datarise** [sqlalchemy/sqlalchemy#8873] [Usage Questions]
Hi, I was wondering if it possible to extract the raw SQL from a model object and print it?

An example could be the following code:

```
class Test(Base):
    __tablename__ = "test"

    id = Column(Integer)
    name= Column(String)

new_test = Test(id=1, name="test")

with Session.begin() as session:
    session.add(new_test) 
    session.commit()
```

Question: 
It is possible to get the raw SQL from the new_test, add or commit and then print it?

**Comment by zzzeek** [comment]
to see the SQL for operations that you're emitting you'd use [echo=True](https://docs.sqlalchemy.org/en/14/core/engines.html#more-on-the-echo-flag)

**Reply by datarise** [reply]
Thanks for the quick reply! 

But is to possible to get the values into a variable? 

`sql_statement = get_sql(new_test) #Example`

**Reply by zzzeek** [reply]
I dont understand what "SQL" is implied by the instance `Test(id=1, name="test")`.

**Reply by datarise** [reply]
I figured it out. Thanks a lot for the help!
